### DISCUSSION TITLE
Future MedusaJS 2.0

### DISCUSSION THREAD

**Discussion by peauc** [medusajs/nextjs-starter-medusa#309] [General]
Hello, after seeing the announcement of MedusaJS 2.0 I've been wondering:
How will the starter handle this version bump :)

**Comment by cmario92** [comment]
what are the breaking changes that expected on MedusaJS 2.0? do we know?

**Comment by VariableVic** [comment]
Hey folks,

As soon as we have a somewhat stable internal release I'm gonna create an updated version of the Starter that works with v2. The plans are not yet set in stone.

As for the breaking changes - all Medusa modules will be fully decoupled. This will change the way you interact with their APIs in some cases. I don't have all the details as of now.

For the Starter update, I expect most of the action to take place in the data fetching layer (eg `src/lib/data/index.ts`) to play well with the fully modular architecture. We might move away from using the JS client in the Starter, as it's unsure if there will be an updated version ready at v2 release.

I'll post an update once things shape up a bit more!

**Reply by peauc** [reply]
Thanks for the update. Will look at closely :eyes:

**Comment by dmdre** [comment]
Hello ! 

I can see that you published the V2 compatible version of this repo ! 

Do you plan to create a migration documentation for this frontend v1 -> v2 ?

Thank you !
